Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java

```diff
@@ -456,6 +481,101 @@ public int compare(String text1, String text2) {
         }
     }
 
+    private void importData(Uri uri) {
+        try {
+            String exportContents = readFile(uri);
+            switch (getFileExtension(uri)) {
+                case "json":
+                    importJsonExport(exportContents);
+                    break;
+                case "txt":
+                    importPlaintextFile(exportContents);
+                    break;
+                case "md":
+                    importPlaintextFile(exportContents);
+                    break;
+                default:
+                    Toast.makeText(requireContext(), getString(R.string.import_unknown), Toast.LENGTH_SHORT).show();
+                    return;
+            }
+            Toast.makeText(requireContext(), getString(R.string.import_message_success), Toast.LENGTH_SHORT).show();
+        } catch (Exception e) {
+            Toast.makeText(requireContext(), getString(R.string.import_message_failure), Toast.LENGTH_SHORT).show();
+        }
```

Comment:
this catch is uncomfortably broad. can we separate the exceptions from `readFile` from the exceptions from the importers? additionally, we have an `AppLog` that would be useful here for debugging issues that appear during the import. if you aren't sure how to add that we can do that. let's add a log when starting the import, indicating which type, and when it succeeds and fails. it would either fit as type `ACTION` or we might create a new type, `IMPORT`

Reply:
I add exception handling. I think it's better to add the logs to you, because I don't know how to do it most correctly.

Reply:
we'll take care of them, thanks

Reply:
Good, thank you too!